cloudinit/sources/DataSourceScaleway.py

```diff
@@ -177,51 +186,152 @@ def __init__(self, sys_cfg, distro, paths):
         self.ds_cfg = util.mergemanydict(
             [
                 util.get_cfg_by_path(sys_cfg, ["datasource", "Scaleway"], {}),
-                BUILTIN_DS_CONFIG,
             ]
         )
 
-        self.metadata_address = self.ds_cfg["metadata_url"]
-        self.userdata_address = self.ds_cfg["userdata_url"]
-        self.vendordata_address = self.ds_cfg["vendordata_url"]
-
         self.retries = int(self.ds_cfg.get("retries", DEF_MD_RETRIES))
         self.timeout = int(self.ds_cfg.get("timeout", DEF_MD_TIMEOUT))
+        self.alive = int(self.ds_cfg.get("alive", DEF_MD_ALIVE))
         self._fallback_interface = None
         self._network_config = sources.UNSET
+        self.metadata_urls = DS_BASE_URLS
+        self.userdata_url = None
+        self.vendordata_url = None
+        self.ephemeral_fixed_address = None
+        self.has_ipv4 = True
+        if "metadata_urls" in self.ds_cfg.keys():
+            # Merge and deduplicate lists
+            self.metadata_urls = list(
+                set(self.metadata_urls) | set(self.ds_cfg["metadata_urls"])
```

Comment:
Sets are, by definition, unordered. Since the list that gets created with the combined set defines the order that urls are checked, we introduce uncertainty into the order that addresses are tried.

Do we really have no preference which address is first? I would think this is undesirable for debugging errors at the very least, and we may actually want to have ipv6 addresses if no other reason than because of its continued adoption / future.

Reply:
I do agree with you that keeping the `metadata_urls` list ordered is most important. The idea behind this was to avoid trying an unreachable url more than once. But now that `url_helper.wait_for_url` is being used, it is no longer a problem since they are tried in parallel.
I'll change this to simply add to the end of the list.